### ISSUE TITLE
NLP parts of speech configuration options 

### ISSUE THREAD

**Issue by bigman73** [RasaHQ/rasa_nlu#555]
**rasa NLU version** (e.g. `0.7.3`):
0.9.0 (docker image)

**Used backend / pipeline** (`mitie`, `spacy_sklearn`, ...):
spacy_sklearn

**Operating system** (windows, osx, ...):
Linux CentOS

**Issue**:
Rasa works with our large training set but seems to be very sensitive to words such as 'that', 'this', 'the' etc.
Is there a way to limit the training and parsing to focus only on certain parts of speech, specifically nouns and verbs?

**Comment by wrathagom** [comment]
Just out of curiosity what are some of your nouns and verbs, have you verified that they are in the corpus?

**Comment by tmbo** [comment]
Which part is sensitive, intent recognition or entity extraction?

**Comment by bigman73** [comment]
@tmbo I'm referring to the intent recognition/classification. 

Entity extraction is another discussion - Rasa currently returns a mixture for intent and entities that are not necessary defined for that intent. Furthermore the entities are mandated to match, which makes no sense. An intent with 2 entities cannot be selected as an absolute winner if the 2 entities are not extracted.

**Comment by tmbo** [comment]
Yes, that makes sense. Actually, there is nothing that prevents us from using the results from the entity extraction for the intent recognition (using the found entities as a feature for the intent extraction - making certain intents more / less likely if entities are present/absent). 

We can only do it one way though, either use the extracted entities for the intent classification or the labeled intent to support the entity extraction. I think this is a very good idea, not sure though if one of them if preferable or if there are use cases for both of them. The implementation for either of them is straight forward.

**Comment by bigman73** [comment]
Thanks. But the entity extraction logic is not the reason I opened this ticket. In fact, I added the entity extraction logic to my post processing logic in the application. 

My current problem with Rasa is that it seems to classify intents in a non NLP way. That is, the machine learning algorithm below is not handling the words in the input sentence in a NLP specific way.
The words 'that' and 'this' have close to zero meaning when classifying intents. They are not the significant topics of the intents. Another example are the different forms in which nouns and verbs can appear. Yes, training can expanded to include many variations and permutations but Rasa immediately falls short when ever a new unrecognized permutation appears.
In fact, if I pass only nouns and verbs to Rasa it would do a better job of classifying. 
From this perspective, it is not a drop in replacement to api.ai, which seems to have excellent NLP and linguistics capabilities. It takes very few training utterances to get the intent to classify against new types of permutations.

**Comment by wrathagom** [comment]
I think the MITIE backend performs similar to API.ai

Do you have a dataset that we can quantify this difference on? A train set that we can use on API.ai, Rasa Spacy, and Rasa MITIE plus a test set to test how each of them performs? Until then I don't see this conversation as being very productive.

There was an issue here recently where someone shared a test that they had ran using API.ai's small talk domain and Rasa was 94% accurate testing against the training set and that was using Spacy_sklearn. #547
